main.tf

```diff
@@ -27,16 +27,16 @@ resource "aws_sns_topic" "this" {
 }
 
 resource "aws_sns_topic_subscription" "this" {
-  for_each = local.enabled ? var.subscribers : {}
+  for_each = { for subscription in toset(var.subscribers) : subscription.endpoint => subscription if local.enabled }
 
   topic_arn              = join("", aws_sns_topic.this.*.arn)
-  protocol               = var.subscribers[each.key].protocol
-  endpoint               = var.subscribers[each.key].endpoint
-  endpoint_auto_confirms = var.subscribers[each.key].endpoint_auto_confirms
-  raw_message_delivery   = var.subscribers[each.key].raw_message_delivery
+  protocol               = each.value.protocol
+  endpoint               = each.value.endpoint
+  endpoint_auto_confirms = lookup(each.value, "endpoint_auto_confirms", false)
+  raw_message_delivery   = lookup(each.value, "raw_message_delivery", false)
```

Comment:
Why set these defaults?

Reply:
moot at this point but I was basing these defaults off of the comment text that accompanies the input variable.